### PR TITLE
nixos/sshguard: use nftables backend if enabled

### PR DESCRIPTION
**Pull request by abrenk**
###### Motivation for this change

I use nftables instead of the default iptables firewall and would like to use sshguard. But the iptables/ipset backend is used unconditionally.

###### Things done

Adapted the module to use the nftables backend if `config.networking.nftables.enable` is `true`.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @peterhoeg @wahjava @danderson

### CODE REVIEW

**Comment by peterhoeg** [comment]
This is great! I will not have time to try it out these next few days but if somebody else does, that would be great.

**Comment by wahjava** [comment]
@abrenk --- I just tried your changes, and it works as expected with my `nftables` setup.

Thanks!

File: nixos/modules/services/security/sshguard.nix [hunk 1]
```diff
@@ -104,22 +107,24 @@ in {
...
+        else [ iptables ipset iproute systemd ];
 
       # The sshguard ipsets must exist before we invoke
       # iptables. sshguard creates the ipsets after startup if
       # necessary, but if we let sshguard do it, we can't reliably add
       # the iptables rules because postStart races with the creation
       # of the ipsets. So instead, we create both the ipsets and
       # firewall rules before sshguard starts.
-      preStart = ''
+      preStart = optionalString config.networking.firewall.enable ''
```
Comment (Profpatsch) [inline nixos/modules/services/security/sshguard.nix]: This change looks unrelated? Or is it required?

**Comment by Profpatsch** [comment]
One small comment, otherwise I’d say we can merge. Sorry this was never merged.

File: nixos/modules/services/security/sshguard.nix (hunk 1 above)
Comment (abrenk) [reply nixos/modules/services/security/sshguard.nix]: It is related and required. `iptables` and `ipset` are not installed when `nftables` is used. The preStart script wouldn't work or make sense as it is not needed to prepare any ipsets or iptables rules.

Comment (Mic92) [reply nixos/modules/services/security/sshguard.nix]: Makes perfect sense to me.
